### DISCUSSION TITLE
Automatic adoption of formatting for automatic theme change on MacOS

### DISCUSSION THREAD

**Discussion by ZimTom** [zim-desktop-wiki/zim-desktop-wiki#2493] [Ideas]
Hi there,
may be the idea changes to a Q&A thing after somebody read this who is more into the details of ZIM.
A couple of months ago I have switched to Apple and I am a great fan of the automatic theme changing depending on lighting conditions.
May be it's sounds like a "luxury problem", but there are two things that I would like to change - or see changed - if that is possible:

1. If changing from day to night mode the system GUI as well as the app's GUI change follow automatically, while ZIM's GUI does not change automatically. The only way to make ZIM change is to quit and to restart and even this sometimes does not help.

2. Even if ZIM changes to dark mode there is a problem with readabilty of links and underlined text.

- Normal Text is shown black on white In day mode, while night mode is using white on dark grey, which is perfect.
- Links are shown blue on white in day mode, which ist fine, but in night mode the foreground color remains unchanged, so that the dark blue on dark grey is really difficult to read.
- Underlined text is shown black on yellow, which is o.k. in terms of readybility, although I would prefer really underlined text. In night mode the foreground color changees to white like the normal text but the yellow background remains making underlined text virtually unreadable.

I hope I have not been blind, but up to know I couldn't find a way to correct this by any means.
May be someone out there has an idea or would this possibly worth a proposal for future versions.

Thanks in advance. Any ideas, hints,... welcome

**Comment by dehesselle** [comment]
### switching between light and dark mode
Under the hood, light and dark mode are different GTK themes. Specific to the macOS app, during startup it checks if macOS is currently in light or dark mode and presets the GTK theme accordingly. This is a pretty simple mechanism and the reason why the theme doesn't change while the program is running. There's also a way to manually set a specific theme to get rid of that automatic switch during startup.
I can only assume that switching themes is possible programmatically at runtime, but that requires someone to research and code this.

### readability of dark theme
I am aware that the dark theme (it's `Adwaita:dark`) has readability issues. This can be fixed, GTK themes are CSS based and can be customized. It would be very welcome if someone with interest and an eye for that kind of work would design a better dark theme. (Basically any GTK3 program can benefit from that.)
It may also be possible that someone has already done this work, because sites like [GNOME-look](https://www.gnome-look.org/browse/) exist.

**Reply by ZimTom** [reply]
Wow, that was fast 🙏, thanks !
compared with people like you I am lightyears away from programming, so things like GTK, CSS etc. are nothing that I would typically see as something I would be able to modify strqight away. But may be I'll give it a little search around and try ... and most likely error 😃.

If the readablity issues in dark mode could be fixed by altering some of the settings in a theme, that would solve the main problem for me and possibly others as well. Restarting ZIM to make it switch between themes is not as much of an issue at the moment, because after quitting the program it restarts on the last page I had opened and even the cursor is at the same spot.

Thanks so far for your swift reaction and the explanation. Liek always here I have learned something new.

Best reagrds
Thomas

**Reply by dehesselle** [reply]
Completely accidental, I have just found a Python-based GTK app that does light/dark mode switching on the fly by clicking a checkbox: https://meld.app

**Comment by dehesselle** [comment]
Closing. Zim 0.76.0 can switch between light and dark mode via checkbox in the settings.
